### PR TITLE
#444 Add Winston Papertrail

### PR DESCRIPTION
**Pull request by simison**

### CODE REVIEW

File: config/lib/logger.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+'use strict';
+// A general purpose logging service that can be used anywhere in the app
+
+/**
+ * Module dependencies.
+ */
+var path = require('path'),
+    _ = require('lodash'),
+    winston = require('winston'),
+    paperTrail = require('winston-papertrail').Papertrail,
```
Comment (simison) [inline config/lib/logger.js]: @chmac Is this needed? At least it's not used anywhere.

File: modules/core/server/jobs/send-email.server.job.js [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+  log('debug', 'Starting `send email` job #wGcxmQ', {jobId});
+
   smtpTransport.sendMail(job.attrs.data, function(err) {
     smtpTransport.close(); // close the connection pool
-    if (err) return done(err);
-    done();
+
+    if (err) {
+      // Log the failure to send the message
+      log('error', 'The `send email` job failed #VDKMbr', {jobId, err})
```
Comment (simison) [inline modules/core/server/jobs/send-email.server.job.js]: @chmac is `#VDKMbr` a tag for Papertrail?

File: config/lib/logger.js (hunk 1 above)
Comment (chmac) [inline config/lib/logger.js]: @simison Which? `paperTrail`? It is used...

Comment (simison) [inline config/lib/logger.js]: Just eslint complains:
```
config/lib/logger.js
  10:5  error  'paperTrail' is defined but never used  no-unused-vars
```

I'll move it out from assigning to a variable.

File: modules/core/server/jobs/send-email.server.job.js [hunk 3]
```diff
@@ -1,14 +1,33 @@
...
+    log = require(path.resolve('./config/lib/logger')).log;
 
 module.exports = function(job, done) {
   var smtpTransport = nodemailer.createTransport(config.mailer.options);
+
+  // The agenda docs don't show the shape of `job` so we're guessing here...
+  var jobId = _.get(job, '_id');
+
+  // Log that we're sending an email
+  log('debug', 'Starting `send email` job #wGcxmQ', { jobId });
```
Comment (simison) [inline modules/core/server/jobs/send-email.server.job.js]: @chmac 
>Can I use `{foo}` where `foo = 'bar'` to create a `{foo: 'bar'}` in es5?

Nope. :-)

ESLint:
```
modules/core/server/jobs/send-email.server.job.js
  16:61  error  Parsing error: Unexpected token }
```

**Comment by simison** [comment]
`npm start` yelds:

```
TypeError: transport is not a constructor
    at Logger.add (/app/node_modules/winston/lib/winston/logger.js:481:41)
    at Object.winston.(anonymous function) [as add] (/app/node_modules/winston/lib/winston.js:86:34)
    at Object.<anonymous> (/app/config/lib/logger.js:24:11)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/app/modules/core/server/jobs/send-email.server.job.js:7:11)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at .<anonymous> (/app/config/lib/worker.js:16:7)
    at emitNone (events.js:86:13)
    at emit (events.js:185:7)
    at handleLegacyCreateIndex (/app/node_modules/agenda/lib/agenda.js:128:10)
[Worker] Script crashed.
```

**Comment by simison** [comment]
@chmac: same same:

```
TypeError: transport is not a constructor
    at Logger.add (/app/node_modules/winston/lib/winston/logger.js:481:41)
    at Object.winston.(anonymous function) [as add] (/app/node_modules/winston/lib/winston.js:86:34)
    at Object.<anonymous> (/app/config/lib/logger.js:21:11)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/app/modules/core/server/jobs/send-email.server.job.js:7:11)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at .<anonymous> (/app/config/lib/worker.js:16:7)
    at emitNone (events.js:86:13)
    at emit (events.js:185:7)
    at handleLegacyCreateIndex (/app/node_modules/agenda/lib/agenda.js:128:10)
[Worker] Script crashed.
```

**Pull request merged**